audit-service/src/unit-tests/java/uk/gov/gchq/palisade/service/audit/service/AuditServiceAsyncProxyTest.java

```diff
@@ -17,68 +17,65 @@
-import uk.gov.gchq.palisade.service.audit.model.AuditSuccessMessage;
+import uk.gov.gchq.palisade.service.audit.model.AuditMessage;
 
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static uk.gov.gchq.palisade.service.audit.ApplicationTestData.auditErrorMessage;
+import static uk.gov.gchq.palisade.service.audit.ApplicationTestData.auditSuccessMessage;
+import static uk.gov.gchq.palisade.service.audit.service.ServiceName.DATA_SERVICE;
+import static uk.gov.gchq.palisade.service.audit.service.ServiceName.FILTERED_RESOURCE_SERVICE;
+import static uk.gov.gchq.palisade.service.audit.service.ServiceName.RESOURCE_SERVICE;
+import static uk.gov.gchq.palisade.service.audit.service.ServiceName.USER_SERVICE;
 
 class AuditServiceAsyncProxyTest {
 
+    private static final List<Boolean> TRUE_RESULT = Collections.singletonList(true);
+    private static final List<Boolean> FALSE_RESULT = Collections.singletonList(false);
+    private static final String TOKEN = "token";
+
     private AuditServiceAsyncProxy proxy;
     private final SimpleAuditService auditService = new SimpleAuditService();
     private final Map<String, AuditService> auditServiceMap = new HashMap<>();
 
-
     @BeforeEach
     void setup() {
         auditServiceMap.put("simple", auditService);
         proxy = new AuditServiceAsyncProxy(auditServiceMap);
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"data-service", "filtered-resource-service"})
-    void testGoodSuccessMessage(final String serviceName) {
-        // Given an Audit Success Message
-        AuditSuccessMessage message = ApplicationTestData.auditSuccessMessage(serviceName);
-
-        // When processed by the proxy service
-        List<Boolean> result = proxy.audit("token", message).join();
-
-        // Then check the returned value
-        assertThat(result).isEqualTo(Collections.singletonList(true));
+    private static Stream<Arguments> messages() {
+        return Stream.of(
+            arguments("good success", USER_SERVICE, auditErrorMessage(USER_SERVICE), TRUE_RESULT),
+            arguments("good success", DATA_SERVICE, auditSuccessMessage(DATA_SERVICE), TRUE_RESULT),
+            arguments("bad success", RESOURCE_SERVICE, auditSuccessMessage(RESOURCE_SERVICE), FALSE_RESULT),
+            arguments("error message", FILTERED_RESOURCE_SERVICE, auditSuccessMessage(FILTERED_RESOURCE_SERVICE),
+                TRUE_RESULT));
     }
 
-    @Test
-    void testBadSuccessMessage() {
-        // Given an Audit Success Message
-        AuditSuccessMessage message = ApplicationTestData.auditSuccessMessage("resource-service");
+    @DisplayName("Test proxy result")
+    @ParameterizedTest(name = "{index} => Test {0}, serviceName=''{1}'', expectedResult={3}")
```

Comment:
what happens to the {2}nd parameter in the arguments?

Reply:
That was a bit of an experiment. When you run it in the ide, the name is what you see. It's actually pretty neat. But it does mean there are extra argument elements just for the display. `text` and `serviceName` are just used in the display, but you have to include all 4 in the method signature. This is what you see:
![image](https://user-images.githubusercontent.com/68223440/109201878-3e5ec800-779a-11eb-9c1e-325e37b5da30.png)